service/popm/popm.go

```diff
@@ -50,6 +50,108 @@ func init() {
 	loggo.ConfigureLoggers(logLevel)
 }
 
+// L2KeystonePriorityBuffer holds up to "size" L2Keystones. it allows a caller
+// to push to it and it will keep the most recent "size" keystones, overriding
+// the oldest if full.  we define "oldest" as "the smallest l2 block number"
+type L2KeystonePriorityBuffer struct {
+	mtx     sync.Mutex
+	mapping map[string]*L2KeystonePriorityBufferElement
+	size    int
+}
+
+// L2KeystonePriorityBufferElement holds an L2Keystone and whether it has been
+// "processed" or not
+type L2KeystonePriorityBufferElement struct {
+	l2Keystone         hemi.L2Keystone
+	requiresProcessing bool
```

Comment:
hmm we might want to add a "currentlyProcessing" flag here as well; since this can be triggered by both timer and new keystone